sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/WriteToDataSourceV2Exec.scala

```diff
@@ -55,6 +57,19 @@ case class WriteToDataSourceV2(
     copy(query = newChild)
 }
 
+object WriteToDataSourceV2Exec {
+  // Returns None if it detects table created by a concurrent command and ifNotExists is true.
+  def handleConcurrentCreateExceptions[T](ifNotExists: Boolean)(block: => T): Option[T] = {
+    try {
+      Some(block)
+    } catch {
+      // Committing to the table failed because of a concurrent table creation to catalog,
+      // but we ignore the exception when IF NOT EXISTS is true.
+      case _: TableAlreadyExistsException if ifNotExists => None
```

Comment:
Is there any log for this silented exception so that user can understand what happens behind?